openpathsampling/ensemble.py

```diff
@@ -253,6 +253,12 @@ def __init__(self):
     def __eq__(self, other):
         if self is other:
             return True
+        if (
+            self.__class__ == other.__class__
+            and self.__uuid__ == other.__uuid__
+        ):
+            return True
+
         return str(self) == str(other)
```

Comment:
Do you actually still want to check the string representation if the UUID fails?

Reply:
Backward compatibility.

This is something that needs to change in OPS 2.0. See https://github.com/openpathsampling/openpathsampling/issues/1080. (Proposal 2 in that issue is similar to what we did with [`gufe`](https://github.com/OpenFreeEnergy/gufe), more docs [here](https://gufe.readthedocs.io/en/latest/guide/serialization.html). I've been keeping an eye on that to see what headaches it causes; there's some conceptual weirdness in the flyweight model, so I'm waiting to see if users hate it.)

Reply:
Also, there's this, which would break:

```python
In [1]: import openpathsampling as paths

In [2]: e1 = paths.EmptyEnsemble()

In [3]: e2 = paths.EmptyEnsemble()

In [4]: e1 == e2  # should be True!
Out[4]: True

In [5]: e1.__uuid__ == e2.__uuid__
Out[5]: False
```